hs_core/page_processors.py

```diff
@@ -166,7 +161,13 @@ def get_page_context(page, user, resource_edit=False, extended_metadata_layout=N
         abstract = content_model.metadata.description.abstract if \
             content_model.metadata.description else None
 
-        missing_metadata_elements = content_model.metadata.get_required_missing_elements()
+        # TODO: #4871 do we think this is a good approach,
```

Comment:
Since we already have `metadata_status` as a context variable, I don't see any issue adding additional context variables as part of the page processor. Otherwise, we can add new property/methods the `CoreMatadata `class (`metadata.get_missing_for_publication() `and` metadata.get_missing_for_discoverable() `- these will be just wrapper methods for the on you already have)) and use those methods directly in the template.

Reply:
Sounds good, thank you @pkdash! Appreciate you taking the time to make a recommendation and steer me in the right direction!